dev/dev-env-setup.sh

```diff
@@ -0,0 +1,254 @@
+
+  - Install pyenv if not installed
+  - Retrieve the appropriate Python version (minimum required) for compatibility support
+  - Check if the virtual environment already exists
+    - If it does, prompt for replacement
+      - If replacing, delete old virtual environment.
+  - Create a virtual environment using the minimum required Python version based on previous step logic
+  - Activate the environment
+  - Install required dependencies for the dev envrionment
+
+  Example usage:
+
+  From root of MLflow repository on local with a destination virtualenv path of <MLFLOW_HOME>/.venvs/mlflow-dev:
+
+  dev/dev-env-setup.sh -d $(pwd)/.venvs/mlflow-dev
+
+  Note: it is recommended to preface virtualenv locations with a directory name prefaced by '.' (i.e., ".venvs").
+
+  The default environment setup is for basic functionality, installing the minimum development requirements dependencies.
+  To install the full development environment that supports working on all flavors and running all tests locally, set
+  the flag '-f' or '--full'
+
+-h, -help,        --help        Display help
+
+-d, -directory    --directory   The path to install the virtual environment into
+
+-f, -full         --full        Whether to install all dev requirements (Default: false)
+
+-v, -verbose      --verbose     Whether to fill stdout with every command or not
+
+-q, -quiet        --quiet       Whether to have pip install in quiet mode (Default: false)
+
+EOF
+}
+
+export verbose=0
+export quiet=0
+while getopts "d:fvqh" opt
+do
+  case "$opt" in
+    d) directory="$OPTARG" ;;
+    f) full_install=1 ;;
+    v) verbose=1 ;;
+    q) quiet=1 ;;
+    h) showHelp; exit ;;
+    *) showHelp; exit ;;
+  esac
+done
+
+if [[ $verbose == 1 ]]; then
+  set -exv
+fi
+
+# Acquire the OS for this environment
+case "$(uname -s)" in
+  Darwin*)                       machine=mac;;
+  Linux*)                        machine=linux;;
+  CYGWIN*|MINGW32*|MSYS*|MINGW*) machine=win;;
+  *)                             machine=unknown;;
+esac
+
+function quietpip(){
+  echo $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' )
+}
+
+# Check if pyenv is installed and offer to install it if not present
+pyenv_exist=$(command -v pyenv)
+
+if [ -z "$pyenv_exist" ]; then
+  if [ -z "$MLFLOW_DEV_ENV_PYENV_INSTALL" ]; then
+    read -p "pyenv is required to be installed to manage python versions. Would you like to install it? $(tput bold)(y/n)$(tput sgr0): " -n 1 -r
+    echo
+  fi
+  if [[ $REPLY =~ ^[Yy]$ || $MLFLOW_DEV_ENV_PYENV_INSTALL == 1 ]]; then
+    if [[ "$machine" == mac ]]; then
+      # Check if brew is installed and install it if it isn't present
+      # Note: if xcode isn't installed, this will fail.
+      if [ -z "$(command -v brew)" ]; then
+        echo "Brew is required to install pyenv on MacOS. Installing in your home directory."
+        wget -O ~/brew_install.sh https://raw.githubusercontent.com/Homebrew/install/master/install.sh
+        bash ~/brew_install.sh
```

Comment:
```suggestion
        bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

to avoid leaving `brew_install.sh`.

Reply:
thanks! Added!